orchestra/utils/task_lifecycle.py

```diff
@@ -819,27 +819,25 @@ def previously_completed_task_data(task):
             A dict mapping task prerequisites onto their latest task
             assignment information.
     """
-    step = task.step
-    prerequisites = {}
+    # Find all prerequisite steps in graph
+    to_visit = list(task.step.creation_depends_on.all())
+    prerequisite_steps = set(to_visit)
+    while to_visit:
+        current_step = to_visit.pop()
+        for step in current_step.creation_depends_on.all():
+            if step not in prerequisite_steps:
+                to_visit.append(step)
+                prerequisite_steps.add(step)
+
+    data = {}
+    for step in prerequisite_steps:
+        required_task = Task.objects.get(step=step, project=task.project)
 
-    for required_step in step.creation_depends_on.all():
-        required_task = Task.objects.get(step=required_step,
-                                         project=task.project)
         if required_task.status != Task.Status.COMPLETE:
             raise TaskDependencyError('Task depenency is not satisfied')
-
-        task_details = get_task_details(required_task.id)
-
         assignment = assignment_history(required_task).last()
```

Comment:
if tests pass, i think it's ok---especially since the implementation of the function is now a lot closer to its docstring :)

assuming the sanity checks in my comments hold true, then :shipit: